### PR TITLE
Do function sanitization logic in setFunction()

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Do function sanitization logic in setFunction()
- Tweak isValidOwnerConfig()'s invalidPath generation logic 

Related issue: https://github.com/ainblockchain/ain-blockchain/issues/413

### CODE REVIEW

File: db/state-util.js [hunk 1]
```diff
@@ -150,42 +152,117 @@ function isValidJsObjectForStates(obj) {
...
     return null;
   }
-  const funcChangeMap = ChainUtil.getJsObject(functionChange, [FunctionProperties.FUNCTION]);
-  if (!funcChangeMap || Object.keys(funcChangeMap).length === 0) {
-    return curFunction;
+
+  const functionType = functionInfo[FunctionProperties.FUNCTION_TYPE];
+  const sanitized = {};
+  if (functionType === FunctionTypes.NATIVE) {
+    sanitized[FunctionProperties.FUNCTION_TYPE] = ChainUtil.stringOrEmpty(functionType)
```
Comment (liayoo) [inline db/state-util.js:163]: could this be just `functionType` or `FunctionTypes.NATIVE`?

File: db/state-util.js [hunk 2]
```diff
@@ -150,42 +152,117 @@ function isValidJsObjectForStates(obj) {
...
+    sanitized[FunctionProperties.FUNCTION_ID] =
+        ChainUtil.stringOrEmpty(functionInfo[FunctionProperties.FUNCTION_ID]);
+  } else if (functionType === FunctionTypes.REST) {
+    sanitized[FunctionProperties.FUNCTION_TYPE] = ChainUtil.stringOrEmpty(functionType)
```
Comment (liayoo) [inline db/state-util.js:167]: ditto

**Review by liayoo — APPROVED** [review]
LGTM with minor comments. Thanks!

File: db/state-util.js (hunk 1 above)
Comment (platfowner) [reply db/state-util.js]: Done. Thanks!

File: db/state-util.js (hunk 2 above)
Comment (platfowner) [reply db/state-util.js]: Done, thanks!

**Pull request merged**